src/components/CountryPicker/index.js

```diff
@@ -0,0 +1,86 @@
+import React, {useState} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../../styles/styles';
+import MenuItemWithTopDescription from '../MenuItemWithTopDescription';
+import * as PersonalDetails from '../../libs/actions/PersonalDetails';
+import useLocalize from '../../hooks/useLocalize';
+import CountrySelectorModal from './CountrySelectorModal';
+import FormHelpMessage from '../FormHelpMessage';
+
+const propTypes = {
+    /** Form Error description */
+    errorText: PropTypes.string,
+
+    /** Country to display */
+    // eslint-disable-next-line react/require-default-props
+    value: PropTypes.string,
+
+    /** Callback to call when the input changes */
+    onInputChange: PropTypes.func,
+
+    /** A ref to forward to MenuItemWithTopDescription */
+    forwardedRef: PropTypes.oneOfType([PropTypes.func, PropTypes.shape({current: PropTypes.instanceOf(React.Component)})]),
+};
+
+const defaultProps = {
+    forwardedRef: undefined,
+    errorText: '',
+    onInputChange: () => {},
+};
+
+function CountryPicker({value: countryValue, errorText, onInputChange, forwardedRef}) {
+    const {translate} = useLocalize();
+    const [isPickerVisible, setIsPickerVisible] = useState(false);
+
+    const showPickerModal = () => {
+        setIsPickerVisible(true);
+    };
+
+    const hidePickerModal = () => {
+        setIsPickerVisible(false);
+    };
+
+    const updateCountryInput = (country) => {
+        onInputChange(country.value);
+        hidePickerModal();
+    };
+
+    const title = PersonalDetails.getCountryName(countryValue);
```

Comment:
Let's use `translate('allCountries')` (same way as done with allStates).

We should also remove `PersonalDetails.getCountryName` function.